### ISSUE TITLE
Lazy preserveScroll not work

### ISSUE THREAD

**Issue by Crisis2010** [inertiajs/inertia#229]
Hello! Seems like merge from issue #74 not work

if I passed preserveScroll as closure, it's working like always True, even if function return hardcoded false value.
Also to proof this I tried to print Page variable to console, and that is not executed.

``` 
             this.$inertia
                .post(this.route('message.create', this.ticket.id), this.form, {
                    preserveScroll: (page) => {
                        console.log('page', page);

                        return false;
                    },
                })
```

**Comment by sinnbeck** [comment]
Seems to have been removed again here

https://github.com/inertiajs/inertia/commit/6da241af975fae4f34f72986ca2eca765b18a4af#diff-e25ae5c7cf909bf9488eea033663bce8

**Comment by reinink** [comment]
Fixed via #231.

Note that we pass the `page` object to these callbacks, whereas previously we returned `page.props`. 👍
